### PR TITLE
WIP: Refactored everything into a module

### PR DESCRIPTION
**Pull request by jclin**
@jameswelle, I realized there was shared code as I sat down to write the deployment cleanup script, so everything has been refactored into a module.

If it looks good, I will merge and test against TC tomorrow.

NOTE: the README hasn't been updated, but essentially the functions in `.\Public\` can be consumed.

### CODE REVIEW

**Comment by jclin** [comment]
Mimicked this blog: http://ramblingcookiemonster.github.io/Building-A-PowerShell-Module/

File: Public/New-AzureVMDeployment.ps1 [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        New-AzureResourceGroup -ResourceGroupName $ResourceGroupName -ResourceGroupLocation $ResourceGroupLocation
+
+        New-AzureResourceGroupDeployment -ResourceGroupName $ResourceGroupName -TemplateFile $script:TemplateFile -TemplateParameterFile $script:TemplateParameterFile
+
+        $vmDomainName = Get-VMDomainName $ResourceGroupName
+        Add-VMDomainNameToTrustedHostsList $vmDomainName
+    }
+    catch
+    {
+        Write-Warning -Message "An error occurred for deployment. '$ResourceGroupName' will be deleted..."
```
Comment (jameswelle) [inline Public/New-AzureVMDeployment.ps1]: Should this be `Write-Error` instead of `Write-Warning`?

File: Public/New-AzureVMDeployment.ps1 [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        Remove-AzureResourceGroup $ResourceGroupName
+
+        Write-Output ($_ | Format-List -Force | Out-String)
```
Comment (jameswelle) [inline Public/New-AzureVMDeployment.ps1]: Is this writing the exception? If so, should it be `Write-Error` as well?

File: Public/Remove-AzureVMDeployment.ps1 [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        Login-AzureAccount -UserName $UserName -SecurePassword $securePassword -SubscriptionId $SubscriptionId
+
+        Remove-AzureResourceGroup $ResourceGroupName
+
+        $vmDomainName = Get-VMDomainName $ResourceGroupName
+        Add-VMDomainNameToTrustedHostsList $vmDomainName
+    }
+    catch
+    {
+        Write-Output ($_ | Format-List -Force | Out-String)
```
Comment (jameswelle) [inline Public/Remove-AzureVMDeployment.ps1]: Same here with `Write-Error`?

**Comment by jameswelle** [comment]
I see in EnablePSRemoting.ps1, we are using `Write-Debug`. Is that intended?

File: Public/New-AzureVMDeployment.ps1 (hunk 2 above)
Comment (jclin) [inline Public/New-AzureVMDeployment.ps1]: Yup, it originally was using `Write-Exception`, but PS wasn't exiting with return code 1 until this was changed :dizzy_face: 

And yes, that piping formats the error record. For ex:
```shell
Exception             : System.DivideByZeroException: Attempted to divide by 
                        zero.
TargetObject          : 
CategoryInfo          : NotSpecified: (:) [], DivideByZeroException
FullyQualifiedErrorId : RuntimeException
ErrorDetails          : 
InvocationInfo        : System.Management.Automation.InvocationInfo
ScriptStackTrace      : at <ScriptBlock>, C:\TeamCity\buildAgent\work\d04f25030
                        3b96f14\New-AzureVMDeployment.ps1: line 139
PipelineIterationInfo : {}
PSMessageDetails      : 
```

**Comment by jclin** [comment]
Yup. IIRC, if that script failed, the deployment of the resource group continued, so I used Write-Debug to get output in the logs on the portal.

**Comment by jclin** [comment]
@jameswelle Updated with two wrapper scripts that the TC agents will call.

**Pull request merged**